compparallel.go

```diff
@@ -326,115 +325,131 @@ func getChannelOrErrorParallel[T any](
 	f func(context.Context, routing.Routing) (<-chan T, error),
 	shouldStop func() bool,
 ) (chan T, error) {
-	outCh := make(chan T)
-	errCh := make(chan error)
-	var wg sync.WaitGroup
+	var ready sync.Mutex
+	ready.Lock()
```

Comment:
I think more explanation around the ready lock seems reasonable. Readers can follow the code, but with async code like this it's easy for anyone who comes next to miss something.